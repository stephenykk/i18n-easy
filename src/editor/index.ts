import { flatten } from 'lodash'
import annotation from './annotation'
import completion from './completion'
import extract from './extract'
import definition from './definition'
import refactor from './refactor'
import problems from './problems'
import reference from './reference'
import statusbar from './statusbar'
import reviewComments from './reviewComments'
import { ExtensionModule } from '~/modules'

const m: ExtensionModule = (ctx) => {
  return flatten([
    annotation(ctx),
    completion(ctx),
    extract(ctx),
    refactor(ctx),
    definition(ctx),
    problems(ctx),
    reference(ctx),
    statusbar(ctx),
    reviewComments(ctx),
  ])
}

export default m
