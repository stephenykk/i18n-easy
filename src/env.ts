// @ts-ignore
export const isProd: boolean = process.env.NODE_ENV === 'production'
export const isDev = !isProd
// @ts-ignore
export const isTest: boolean = process.env.NODE_ENV === 'test'
