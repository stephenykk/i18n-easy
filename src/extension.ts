import { ExtensionContext } from 'vscode'
import { flatten } from 'lodash'
import { version } from '../package.json'
import { Global, Config, KeyDetector, CurrentFile } from '~/core'
import commandsModules, { Commands } from '~/commands'
import viewsModules from '~/views'
import { Log } from '~/utils'
import i18n from '~/i18n'
import editorModules from '~/editor'

export async function activate(ctx: ExtensionContext) {
  Log.info(`ðŸˆ¶ Activated, v${version}`)

  Config.ctx = ctx

  i18n.init(ctx.extensionPath)
  KeyDetector.init(ctx)

  // activate the extension
  await Global.init(ctx)
  CurrentFile.watch(ctx)

  const modules = [
    commandsModules,
    editorModules,
    viewsModules,
  ]

  const disposables = flatten(modules.map(m => m(ctx)))
  disposables.forEach(d => ctx.subscriptions.push(d))
}

export function deactivate() {
  Log.info('ðŸˆš Deactivated')
}

export {
  Global,
  CurrentFile,
  KeyDetector,
  Config,
  Log,
  Commands,
}
