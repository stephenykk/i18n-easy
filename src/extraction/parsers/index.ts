import * as html from './html'
import * as babel from './babel'

export const extractionsParsers = {
  html,
  babel,
}
