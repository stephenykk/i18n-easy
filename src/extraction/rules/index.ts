import { BasicExtrationRule } from './basic'
import { NonAsciiExtractionRule } from './non-ascii-characters'
import { DynamicExtractionRule } from './dynamic'

export * from './base'
export * from './basic'
export * from './non-ascii-characters'

export const DefaultExtractionRules = [
  new BasicExtrationRule(),
  new NonAsciiExtractionRule(),
]

export const DefaultDynamicExtractionsRules = [
  new DynamicExtractionRule(),
]
