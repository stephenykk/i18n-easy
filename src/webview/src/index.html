<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf8" />
    <meta name="viewport" content="width=device-width" />
  </head>

  <body>
    <div id="app"></div>
    <script src="./index.js"></script>
  </body>
</html>
