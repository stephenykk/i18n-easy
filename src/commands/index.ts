import { flatten } from 'lodash'
import configLocales from './configLocalePaths'
import configLanguages from './configLanguages'
import keyManipulations from './keyManipulations'
import extractText from './extractString'
import detectHardStrings from './detectHardStrings'
import help from './help'
import refreshUsageReport from './refreshUsageReport'
import editor from './openEditor'
import review from './review'
import deepl from './deepl'
import gotoRange from './gotoRange'
import gotoNextUsage from './gotoNextUsage'
import batchHardStringsExtract from './extractStringBulk'
import { ExtensionModule } from '~/modules'

const m: ExtensionModule = (ctx) => {
  return flatten([
    configLocales(ctx),
    configLanguages(ctx),
    keyManipulations(ctx),
    extractText(ctx),
    detectHardStrings(ctx),
    batchHardStringsExtract(ctx),
    help(ctx),
    refreshUsageReport(ctx),
    editor(ctx),
    review(ctx),
    deepl(ctx),
    gotoRange(ctx),
    gotoNextUsage(ctx),
  ])
}

export * from './commands'

export default m
