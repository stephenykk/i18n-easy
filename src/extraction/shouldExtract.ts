import { DefaultExtractionRules, ExtractionRule, ExtractionScore } from './rules'

export function shouldExtract(str: string, rules: ExtractionRule[] = DefaultExtractionRules): boolean {
  let extract = false
  for (const rule of rules) {
    const result = rule.shouldExtract(str)
    if (result === ExtractionScore.MustExclude)
      return false
    if (result === ExtractionScore.MustInclude)
      return true
    if (result === ExtractionScore.ShouldExclude)
      extract = false
    if (result === ExtractionScore.ShouldInclude)
      extract = true
  }
  return extract
}
