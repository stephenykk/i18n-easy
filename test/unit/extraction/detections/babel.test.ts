import { basename } from 'path'
import { promises as fs } from 'fs'
import fg from 'fast-glob'
import { expect } from 'chai'
import { extractionsParsers } from '../../../../src/extraction'

const babel = extractionsParsers.babel

describe('detections - babel', () => {
  const files = fg.sync('../../../fixtures/vue/scripts/*.*', {
    cwd: __dirname,
    absolute: true,
  })

  for (const file of files) {
    const name = basename(file)

    it(name, async() => {
      const content = await fs.readFile(file, 'utf-8')
      const result = babel.detect(content)
      expect(result.map(i => i.text)).to.matchSnapshot()
    })
  }
})
