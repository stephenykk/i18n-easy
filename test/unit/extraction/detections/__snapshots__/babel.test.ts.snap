// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detections - babel v2-cdn.js 1`] = `
Array [
  "This should be detected",
]
`;

exports[`detections - babel v2-cjs.js 1`] = `
Array [
  "Hello",
]
`;

exports[`detections - babel v2-class.js 1`] = `Array []`;

exports[`detections - babel v2-class-tsx.tsx 1`] = `Array []`;

exports[`detections - babel v2-esm.js 1`] = `
Array [
  "About page",
]
`;

exports[`detections - babel v2-jsx.jsx 1`] = `
Array [
  "Nigeria",
  "Nigeria is a large country that has a varied topography.",
]
`;

exports[`detections - babel v2-jsx-2.jsx 1`] = `
Array [
  "Home page ðŸš€",
  "Home page description",
  "Home page ðŸš€",
  "About page",
]
`;

exports[`detections - babel v2-ts.ts 1`] = `
Array [
  "v-date-picker-table v-date-picker-table--date",
]
`;

exports[`detections - babel v3-ts.ts 1`] = `
Array [
  "Filename",
]
`;

exports[`detections - babel v3-tsx.tsx 1`] = `
Array [
  "AComment",
]
`;
